test/OpenTelemetry.Tests/Logs/LogRecordTest.cs

```diff
@@ -681,6 +736,51 @@ public override void OnEnd(LogRecord data)
                 base.OnEnd(data);
             }
         }
+
+        private class RedactionExporter : BaseExporter<LogRecord>
```

Comment:
Two comments...

* I question the validity of this example. Should an exporter be modifying the data? What if the user has multiple exporters registered? This type of thing should be done in a LogProcessor, no?

* Seems like the user wants to scrub log messages. Writing a custom processor or export is kind of heavy lifting. What if we provided an `Enrich` callback on `OpenTelemetryLoggerOptions` to make this case slightly easier? Could be called here:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/7ce2e4ca2308e7805c54181e61458de24c0c7907/src/OpenTelemetry/Logs/OpenTelemetryLogger.cs#L54-L65

Like...

```csharp
                var record = new LogRecord(
                    options.IncludeScopes ? this.ScopeProvider : null,
                    DateTime.UtcNow,
                    this.categoryName,
                    logLevel,
                    eventId,
                    options.IncludeFormattedMessage ? formatter?.Invoke(state, exception) : null,
                    options.ParseStateValues ? null : (object)state,
                    exception,
                    options.ParseStateValues ? this.ParseState(state) : null);

                // New
                options.Enrich?.Invoke(record);

                processor.OnEnd(record);
```

If we do that callback, we could pass in the `record` instance and then have the "Set" APIs available for the user to call to do the mutation. If we wanted to keep `record` immutable (which I kind of like) we could change the enrich API into something where the raw[ish] log data is passed in and the `LogRecord` instance is returned.

Reply:
My bad, I didn't make it clear on what is the issue I'm trying to address with this PR and I think the test case is misleading. I've created a feature request to explain the goal of this PR here: https://github.com/open-telemetry/opentelemetry-dotnet/issues/2877.

Reply:
I've updated the test case to use Processor as @CodeBlanch as suggested.
The reason why Exporter was used is for performance reason for the specific case of credential scanning. 
Credential scanning involves in regex matching and each regex match can take several microseconds to execute. Since exporter thread could be run asynchronously, API calls will not be blocked.